### PR TITLE
UI changes

### PR DESCRIPTION
**Pull request by bekand**
# Terms

> THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE TERMS OF THE CCLA DATED 2017-11-07 WITH FINOS/LINUX FOUNDATION (FORMERLY THE SYMPHONY SOFTWARE FOUNDATION CCLA).
>
> THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

### CODE REVIEW

File: cli/src/Morphir/Web/DevelopApp.elm [hunk 1]
```diff
@@ -965,23 +970,27 @@ viewHeader model =
...
         [ row
             [ width fill ]
-            [ row
-                [ width fill
-                ]
-                [ el [ padding 6 ]
-                    (image
-                        [ height (px 40)
+            [ link []
+                { url = pathToFullUrl [] ++ filterStateToQueryParams model.homeState.filterState
```
Comment (AttilaMihaly) [inline cli/src/Morphir/Web/DevelopApp.elm:974]: I'm not sure what this means. What's the purpose of the empty list there?

File: cli/src/Morphir/Web/DevelopApp.elm [hunk 2]
```diff
@@ -1344,6 +1353,60 @@ viewHome model packageName packageDef =
...
+                , tabs =
+                    Array.fromList
+                        [ { name = "Summary"
+                          , content = el [ padding <| Theme.smallPadding model.theme ] summary
+                          }
+                        , { name = "Dependency Graph"
+                          , content = col [ dependencyGraph model.homeState.selectedModule model.repo ]
+                          }
+                        ]
+                }
```
Comment (AttilaMihaly) [inline cli/src/Morphir/Web/DevelopApp.elm:1409]: I think this whole section would deserve its own function. Also, make sure you always specify the types of your definitions.

File: cli/src/Morphir/Web/Graph/DependencyGraph.elm [hunk 3]
```diff
@@ -35,42 +38,72 @@ viewGraph graph =
...
     let
-        gray =
-            rgb 0.9 0.9 0.9
-
-        filterDepsBySelectedModule : DAG.DAG FQName -> List ( String, List String )
-        filterDepsBySelectedModule deps =
+        filterDepsBySelectedModule : DAG.DAG FQName -> Maybe ModuleName -> List Dependency
+        filterDepsBySelectedModule deps moduleName =
+            let
+                createDependencyTuple nodeName childNodeNames =
```
Comment (AttilaMihaly) [inline cli/src/Morphir/Web/Graph/DependencyGraph.elm:44]: Can you add the types here?

File: cli/src/Morphir/Web/Graph/DependencyGraph.elm [hunk 4]
```diff
@@ -35,42 +38,72 @@ viewGraph graph =
...
+                                    else
+                                        Nothing
+                                )
+
+                        Nothing ->
+                            List.map
+                                (\( fqName, fqNameSet ) ->
+                                    createDependencyTuple fqName fqNameSet
+                                )
+                   )
```
Comment (AttilaMihaly) [inline cli/src/Morphir/Web/Graph/DependencyGraph.elm:68]: This is very elegant but I think in this situation it's better to have a bit of duplication for readability. I would move the `deps |> DAG.toList` part inside both branches and move the pattern match to the top.

**Review by AttilaMihaly — CHANGES_REQUESTED** [review]

File: src/Morphir/Elm/IncrementalFrontend.elm [hunk 5]
```diff
@@ -577,9 +577,16 @@ processModule moduleName parsedModule opts exposedModules repo =
 
                     else
                         Private
+
+                documentedModule : Module.Definition () (Type ())
```
Comment (KindsonTheGenius) [inline src/Morphir/Elm/IncrementalFrontend.elm:581]: Could you just use an emptyDefinition here instead of creating a new module which is empty anyway.

Comment (bekand) [reply src/Morphir/Elm/IncrementalFrontend.elm:581]: It's not a completely empty module, the doc field is set.

Comment (AttilaMihaly) [reply src/Morphir/Elm/IncrementalFrontend.elm:581]: It's not completely empty, it has docs.

**Review by AttilaMihaly — CHANGES_REQUESTED** [review]
It looks good to me. It can be merged after you fix the codec issue that @KindsonTheGenius pointed out.

File: src/Morphir/Elm/IncrementalFrontend.elm (hunk 5 above)
Comment (KindsonTheGenius) [reply src/Morphir/Elm/IncrementalFrontend.elm:581]: Ok. This is fine.

**Review by AttilaMihaly — APPROVED** [review]

**Pull request merged**